### ISSUE TITLE
textual_attack.py的token_gradient方法在return处出现one_hot.grad为None而报错

### ISSUE THREAD

**Issue by Thunderank** [cure-lab/MMA-Diffusion#1]
![QQ截图20240413232149](https://github.com/cure-lab/MMA-Diffusion/assets/106507105/278d7e3e-948b-4bb4-9117-19f03f380419)
如图所示，经过loss.backward之后one_hot仍为None类型

**Comment by Windy4096** [comment]
hello, do u solve the problem?

**Comment by Thunderank** [comment]
> hello, do u solve the problem?

No,I'm still stuck in here.
I have tested all relevent variables and now suspect that something goes wrong with the "CosineSimilarityLoss" class

**Comment by yangyijune** [comment]
Hi! Have you resolved this issue? I haven't come across this problem myself. It's possible that there might be some gradient blocking codes in your transformers package's clip_modeling.py, such as @torch.no_grad(), which could be affecting the process.

**Comment by 1998v7** [comment]
您好 请问您有碰到过hugg报错API失效吗？

**Comment by 1998v7** [comment]
Have you solved this problem?

**Comment by Thunderank** [comment]
没有，我这边按照readme的步骤配置的环境，使用mma_diffusion_environment.yml安装的依赖，然后也按照readme中的提示修改了transformer源代码添加了属性，但运行的时候报的错就是token_gradient(model, tokenizer, control, target_embeddings)函数中的return one_hot.grad.clone()一行提示AttributeError: 'NoneType' object has no attribute 'clone'，输出发现one_hot对象不包含grad属性。
目前为止仍没有解决这一问题，但也没有遇到过hugg报错API失效的问题。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cure-lab/MMA-Diffusion"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年5月26日(星期天) 中午11:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [cure-lab/MMA-Diffusion] textual_attack.py的token_gradient方法在return处出现one_hot.grad为None而报错 (Issue #1)





 
您好 请问您有碰到过hugg报错API失效吗？

**Comment by tsrigo** [comment]
我刚刚也遇到了这个问题，后面发现是添加属性没有添加完整，总共有四处位置
![image](https://github.com/user-attachments/assets/8f3af7e9-911a-45a0-807d-20f638e6cc49)
